2048Solution/main/java/models/GameService.java

```diff
@@ -0,0 +1,230 @@
+package main.java.models;
+import java.util.ArrayList;
+import java.lang.Integer;
+public class GameService {
+    GameBoard board = new GameBoard();
+    GameUtility gameUtility = new GameUtility();
+    void initialize(int size){
+        board.setSize(size);
+        board.initializeBoard(size);
+        fillValueInBoard();
+        fillValueInBoard();
+    }
+
+    void fillValueInBoard(){
+        gameUtility.getRandomIndex(board);
+        board.setValueInBoard(board.getRowToBeFilled(),board.getColumnToBeFilled(), 2);
+    }
+    void printValues(){
+        for(int i = 0; i < board.getSize(); i++){
+            for(int j = 0; j < board.getSize(); j++){
+                if(board.getValueInBoard(i,j) == 0){
+                    System.out.print("- ");
+                }
+                else{
+                    System.out.print(board.getValueInBoard(i,j) + " ");
+                }
+            }
+            System.out.println();
+        }
+    }
+    void startGame(int direction){
+       makeDecision(direction);
```

Comment:
Not clear what this means

Reply:
makeDecision function contains a switch case which decides which direction to move the grid to.
So, it is a function where only decisions are made

Reply:
Name not sufficient for that. What kind of decision does it make? The name should be sufficient enough to understand the purpose of the function without having to look at the implementation of the function.